drivers/adc/ad4170/ad4170.c

```diff
@@ -0,0 +1,2119 @@
+	if (ret)
+		return ret;
+
+	if (dev->gpio_dig_aux2) {
+		if (dev->dig_aux2_output) {
+			// when dig_aux2 is used as LDAC or SYNC
+			ret = no_os_gpio_direction_output(dev->gpio_dig_aux2, NO_OS_GPIO_HIGH);
+			if (ret)
+				return ret;
+		} else {
+			// when dig_aux2 is used as MODOUT or disabled
+			ret = no_os_gpio_direction_input(dev->gpio_dig_aux2);
+			if (ret)
+				return ret;
+		}
+	}
+
+	ret = no_os_gpio_get_optional(&dev->gpio_sync_inb, dev->gpio_sync_inb_init);
+	if (ret)
+		return ret;
+
+	if (dev->gpio_sync_inb) {
+		ret = no_os_gpio_direction_output(dev->gpio_sync_inb, NO_OS_GPIO_HIGH);
+		if (ret)
+			return ret;
+	}
+
+	ret = no_os_spi_init(&dev->spi_desc, &dev->spi_init);
+	if (ret)
+		return ret;
+
+	dev->digif = true;
+
+	return ret;
+}
+
+static void _ad4170_digif_remove(struct ad4170_dev *dev)
+{
+	if (!dev)
+		return;
+	no_os_spi_remove(dev->spi_desc);
+	no_os_gpio_remove(dev->gpio_dig_aux1);
+	no_os_gpio_remove(dev->gpio_dig_aux2);
+	no_os_gpio_remove(dev->gpio_sync_inb);
+	dev->digif = false;
+}
+
+/**
+ * @brief Exit continuous read mode
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_continuous_read_exit(struct ad4170_dev *dev)
+{
+	int ret;
+	uint32_t data;
+	uint8_t status;
+
+	if (!dev)
+		return -EINVAL;
+
+	if (dev->config.adc_ctrl.cont_read != AD4170_CONT_READ_ON)
+		return -EACCES;
+
+	ret = _ad4170_continuous_read_one(dev, &data, &status, true);
+
+	if (!ret)
+		dev->config.adc_ctrl.cont_read = AD4170_CONT_READ_OFF;
+
+	return ret;
+}
+
+/**
+ * @brief Exit continuous transmit mode
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_continuous_transmit_exit(struct ad4170_dev *dev)
+{
+	int ret;
+	struct ad4170_adc_ctrl adc_ctrl = dev->config.adc_ctrl;
+
+	if (!dev)
+		return -EINVAL;
+
+	if (dev->config.adc_ctrl.cont_read != AD4170_CONT_TRANSMIT_ON)
+		return -EACCES;
+
+	if (!dev->digif) {
+		ret = _ad4170_digif_init(dev);
+		if (ret)
+			return 0;
+	}
+
+	adc_ctrl.cont_read = AD4170_CONT_READ_OFF;
+	ret = ad4170_set_adc_ctrl(dev, adc_ctrl);
+	// Being in continuous transmit (TDM), the slave may put TDM data
+	// on SPI MISO which will make the CRC check fail, if enabled.
+	if (ret && ret != -EBADMSG)
+		return ret;
+
+	dev->config.adc_ctrl.cont_read = AD4170_CONT_READ_OFF;
+
+	return 0;
+}
+
+/**
+ * @brief Perform a software reset
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4170_reset(struct ad4170_dev *dev)
+{
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad4170_spi_reg_write_mask(dev,
+					AD4170_REG_INTERFACE_CONFIG_A,
+					AD4170_SW_RESET_MSK | AD4170_SW_RESETX_MSK,
+					AD4170_SW_RESET_MSK | AD4170_SW_RESETX_MSK);
+	if (ret)
+		return ret;
+
+	/* TODO: set proper value, it's currently undefined in datasheet. */
```

Comment:
Is there any datasheet available now?

Reply:
good point, will try to find the right value

Reply:
found it in the datasheet as Turn-On Settling Time, specified to 200us, so I will update accordingly